### ISSUE TITLE
Fold typescript definition uses potentially pointless generic L

### ISSUE THREAD

**Issue by bagdonas** [fluture-js/Fluture#334]
Correct me if I'm wrong, but fold is defined as:

```hs
fold  :: Future a b ~> (a -> c, b -> c) -> Future d c
```

However in typescript (index.d.ts) fold is defined as:

```ts
fold<L, RB>(lmapper: (reason: L) => RB, rmapper: (value: R) => RB): FutureInstance<L, RB>
```

or

```hs
fold  :: Future a b ~> (a -> c, b -> c) -> Future a c
```

right? I believe the correct definition should be:

```ts
fold<RL, RB>(lmapper: (reason: L) => RB, rmapper: (value: R) => RB): FutureInstance<RL, RB>
//or
fold<RB>(lmapper: (reason: L) => RB, rmapper: (value: R) => RB): FutureInstance<L, RB>
//or
fold<RB>(lmapper: (reason: L) => RB, rmapper: (value: R) => RB): FutureInstance<void, RB>
```

Personally I would prefer the latter one

**Comment by Avaq** [comment]
> I believe the correct definition should be [one of my three suggestions.]

You're absolutely right! This issue has been resolved in the beta version of Fluture though.
However, while looking into it, I found another issue with the new definition of `fold` - I'll fix it straight away.

> Personally I would prefer the latter one

The latter one is not correct, because void is not the same as a yet-to-be-determined type variable. But that's a minor detail. It might be an idea to use [`unknown`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type) instead of `void`.
